### PR TITLE
fix(core): multi-select doesn't work with objects

### PR DESCRIPTION
**Pull request by dmitry-stepanenko**
## Related Issue(s)

closes https://github.com/SAP/fundamental-ngx/issues/7408

## Description
core-multi-select can work only with strings or object references. When the object is cloned, it can no longer be identified as selected. This is because selected state is determined by plain equality (===) and there's no way to pass comparer function.

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [n/a] different combinations of components - free style
-   [n/a] change the API values during testing

3. Documentation and Example validations

-   [n/a] missing API documentation or it is not understandable
-   [x] poor examples
-   [n/a] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   x ] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [n/a] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: libs/core/src/lib/multi-input/multi-input.component.ts [hunk 1]
```diff
@@ -130,6 +143,9 @@ export class MultiInputComponent
     @Input()
     filterFn = this._defaultFilter;
 
+    @Input()
```
Comment (dimamarksman) [inline libs/core/src/lib/multi-input/multi-input.component.ts:146]: could you please add comment description

**Review by dimamarksman — APPROVED** [review]

File: libs/core/src/lib/multi-input/multi-input.component.html [hunk 2]
```diff
@@ -1,143 +1,150 @@
...
+                [fillControlMode]="fillControlMode"
+                class="fd-multi-input-popover-custom"
             >
-                <ng-container *ngTemplateOutlet="list"></ng-container>
-                <ng-content></ng-content>
-            </fd-popover-body>
-        </fd-popover>
+                <fd-popover-control>
+                    <form (submit)="_onSubmit()">
+                        <ng-container>
```
[the remark this answers is not in the document]
Comment (dmitry-stepanenko) [reply libs/core/src/lib/multi-input/multi-input.component.html]: wasn't my change, removed now

File: libs/core/src/lib/multi-input/multi-input.component.ts (hunk 1 above)
Comment (dmitry-stepanenko) [reply libs/core/src/lib/multi-input/multi-input.component.ts:154]: done

**Review by platon-rov — APPROVED** [review]

**Review by Betrozov — APPROVED** [review]

**Comment by maxym-valor** [comment]
No icons and items names in drop-down for Custom Item Template example. 
Expected: 
image
Actual: 
image

**Review by Betrozov — APPROVED** [review]

**Pull request merged**